Enhancements_Fast_api/main.py

```diff
@@ -0,0 +1,248 @@
+        attack=pokemon.attack,
+        defense=pokemon.defense,
+        sp_atk=pokemon.sp_atk,
+        sp_def=pokemon.sp_def,
+        speed=pokemon.speed,
+        generation=pokemon.generation,
+        legendary=pokemon.legendary,
+    )
+    db.add(newPokemon)
+    db.commit()
+    db.refresh(newPokemon)
+    return newPokemon
+
+
+@app.put(
+    "/pokemon/{pokemon_id}",
+    response_model=PokemonPostPatchPutOutputSchema,
+    status_code=status.HTTP_202_ACCEPTED,
+)
+def update_Pokemon(
+    pokemon_id: int, pokemon: PokemonPostPutInputSchema, db: Session = Depends(get_db)
+):
+    find_pokemon = (
+        db.query(models.PokemonData).filter(models.PokemonData.id == pokemon_id).first()
+    )
+    if not find_pokemon:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail="Pokemon with this id does not exist.",
+        )
+
+    find_pokemon.name = pokemon.name
+    find_pokemon.type_1 = pokemon.type_1
+    find_pokemon.type_2 = pokemon.type_2
+    find_pokemon.total = pokemon.total
+    find_pokemon.hp = pokemon.hp
+    find_pokemon.attack = pokemon.attack
+    find_pokemon.defense = pokemon.defense
+    find_pokemon.sp_atk = pokemon.sp_atk
+    find_pokemon.sp_def = pokemon.sp_def
+    find_pokemon.speed = pokemon.speed
+    find_pokemon.generation = pokemon.generation
+    find_pokemon.legendary = pokemon.legendary
+
+    db.commit()
+    return find_pokemon
+
+
+@app.patch("/pokemon/{pokemon_id}", response_model=PokemonPostPatchPutOutputSchema)
+def update_Pokemon_Patch(
+    pokemon_id: int, pokemon: PokemonPatchInputSchema, db: Session = Depends(get_db)
+):
+    find_pokemon = (
+        db.query(models.PokemonData).filter(models.PokemonData.id == pokemon_id).first()
+    )
+    if not find_pokemon:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Pokemon with id {pokemon_id} doesn't exist...",
+        )
+
+    for key, value in pokemon.dict(exclude_unset=True).items():
+        setattr(find_pokemon, key, value)
+
+    db.commit()
+    return find_pokemon
+
+
+@app.delete("/pokemon/{pokemon_id}", status_code=status.HTTP_204_NO_CONTENT)
+def delete_Pokemon(pokemon_id: int, db: Session = Depends(get_db)):
+    find_pokemon = (
+        db.query(models.PokemonData).filter(models.PokemonData.id == pokemon_id).first()
+    )
+
+    if not find_pokemon:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND, detail="Pokemon not found"
+        )
+
+    db.delete(find_pokemon)
+    db.commit()
+    return None
+
+
+@app.post("/pokemon/")
+def fetch_and_store(db: Session = Depends(get_db)):
+    response = requests.get("https://coralvanda.github.io/pokemon_data.json")
+    data = response.json()
+
+    print(f"Data fetched: {len(data)} entries")
+
+    # Fetch the current max ID in the database
+    max_id = db.query(func.max(models.PokemonData.id)).scalar() or 0
+    current_id = max_id + 1
```

Comment:
these two lines (#167, #168) may not be relevant/necessary here, assuming that you are loading the Pokemon data for the first time.

Reply:
I have removed this line (#167, #168) will update the auto increment.